C/emit_log_direct.c

```diff
@@ -0,0 +1,148 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+
+#include <amqp.h>
+#include <amqp_tcp_socket.h>
+
+int main(int argc, char const *const *argv) {
+    char const *severity;
+    char *message;
+    amqp_connection_state_t conn;
+    amqp_socket_t *socket = NULL;
+    const char* DEFAULT_HOSTNAME = "localhost";
+    const int DEFAULT_PORT = 5672;
+    int socket_status;
+    const char* DEFAULT_VHOST = "/";
+    const char* DEFAULT_USER = "guest";
+    const char* DEFAULT_PASSWORD = "guest";
+    amqp_rpc_reply_t rpc_reply;
+    const amqp_channel_t CHANNEL_ID = 1;
+    amqp_channel_open_ok_t* channel_status;
+    const char* EXCHANGE_NAME = "direct_logs";
+    const char* EXCHANGE_TYPE = "direct";
+    amqp_exchange_declare_ok_t* exchange_status;
+    int publish_status;
+
+    // Prepare message
+    if (argc == 1) {
+        severity = "info";
+        message = "Hello World from C producer!";
+    } else if (argc == 2) {
+        severity = argv[1];
+        message = "Hello World from C producer!";
+    } else if (argc > 2) {
+        severity = argv[1];
+
+        int i, v = 0;
+        message = (char*)malloc(v);
```

Comment:
The comments added to `emit_log.c` are also true for this program and all others doing the same kind of concatenation.